CMakeLists.txt

```diff
@@ -42,6 +42,7 @@ endif()
 
 option(SOCI_SHARED "Enable building SOCI as a shared library" ${SHARED_DEFAULT})
 option(SOCI_TESTS "Enable building SOCI test cases" ${PROJECT_IS_TOP_LEVEL})
+option(WITH_BOOST "Enable Boost support" ON)
```

Comment:
I would recommend a name more in line with the new CMake options such as `SOCI_BOOST`. Then add some logic to the compat cmake file that sets the value of `SOCI_BOOST` if `WITH_BOOST` is defined.

Furthermore, I don't think the semantics of this option are quite clear. If we have a ON/OFF option, I would expect an error if it is set to `ON` but Boost is not found. However, it seems like the semantics are more akin to the `AUTO` mode for the backends?

An alternative would be to instead have an option `SOCI_DISABLE_BOOST` that leaves the current auto-detect logic if set to `OFF` and disables if set to `ON`

Reply:
Okay, fair enough